modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/streaming/BytesStreamingExtensionTestCase.java

```diff
@@ -199,6 +201,26 @@ public void streamProviderSerialization() throws Exception {
   }
 
   @Test
+  @Description("A non repeatable stream is closed automatically when flow completes")
+  public void nonRepeatableStreamIsAutomaticallyClosed() throws Exception {
+    InputStream stream = (InputStream) flowRunner("toNonRepeatableStream")
+        .withPayload(data).run().getMessage().getPayload().getValue();
+    assertThat(stream.read(), is(-1));
+  }
+
+  @Test
+  @Description("A non repeatable stream is not automatically closed when event completes async")
+  public void nonRepeatableStreamClosesAsync() throws Exception {
+    InputStream stream = (InputStream) flowRunner("toNonRepeatableStream").keepStreamsOpen()
+        .withPayload(data).run().getMessage().getPayload().getValue();
+
+    byte[] bytes = new byte[data.length()];
+    assertThat(stream.read(bytes), is(data.length()));
+    assertThat(stream.read(), is(-1));
+    assertThat(new String(bytes), equalTo(data));
+  }
+
```

Comment:
Need to test the error case too.

Reply:
which error case?

Reply:
The case where an error is thrown in a flow and error is handled by an error-handler. This is a different code path and its critical that streams are also closed in this case, so it should really be covered with tests